rust/agama-server/src/web.rs

```diff
@@ -114,6 +114,24 @@ async fn run_events_monitor(dbus: zbus::Connection, events: EventsSender) -> Res
         )
         .await?,
     );
+    stream.insert(
+        "software-issues",
+        issues_stream(
+            dbus.clone(),
+            "org.opensuse.Agama.Software1",
+            "/org/opensuse/Agama/Software1",
+        )
+        .await?,
+    );
+    stream.insert(
+        "software-product-issues",
```

Comment:
similar here. I would either expect some id in issue report that can be used to make it specific or at least use just `product-issues`, otherwise naming looks inconsistent

Reply:
I am not using those IDs for anything, so they aren't relevant. I decided to, somehow, match the D-Bus structure (`/org/opensuse/Agama/Software1` and `/org/opensuse/Agama/Software1/Product`).

However, if we use it as an ID, we could get rid of the `service`, hiding to the client the fact that it is talking to a D-Bus service:

```javascript
WithIssues("software/issues")
```

instead of

```javascript
WithIssues("software/issues", "/org/opensuse/Agama/Software1"
```

In that case, I would adjust the events-related types to something like:

```rust
pub type EventsScope = String;
pub type EventsSender = Sender<(Scope, Event)>;
pub type EventsReceiver = Receiver<(Scope, Event)>;
```

Does it make sense?

Reply:
On the other hand, keeping the scope out of the event introduces some small inconsistencies, like being able to send a `LocaleChanged` event from any unrelated scope. But I guess I can live with that.

I have been also thinking about using something like:

```rust
pub enum Event {
  Locale(LocaleEvent),
  Software(SoftwareEvent),
};
```

And then each service defines its event types:

```rust
pub enum LocaleEvent {
  LocaleChanged { locale: String },
  IssuesChanged(IssuesChangedEvent),
  ProgressChanged(ProgressChangedEvent)
}
```

In this case, the scope is implicit.

I would like to think a little bit about that (and other options we could think of). For now, I will just update the IDs to make them look better and more consistent.

Reply:
makes sense to me to think a bit about it and what is more important document decision, so we will be consistent.

Reply:
Just a note about the second proposal: if you can emit a `IssuesChange` in several places from a single service, you still need a way to differentiate them.